persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/RestoreBackupTest.kt

```diff
@@ -0,0 +1,487 @@
+        val restoredUsers = userDatabaseBuilder.userDAO.getAllUsers().first()
+        assertEquals(restoredUsers,  uniqueUsers + uniqueBackupUsers)
+        assertEquals(uniqueUsersAmount + uniqueBackupUsersAmount, restoredUsers.size)
+    }
+
+    @Test
+    fun givenBackupHasAssetsButUserNot_whenRestoringBackup_thenBackupAssetsAreRestored() = runTest {
+        // given
+        val assetsToBackup = backupDatabaseDataGenerator.generateAndInsertAssets(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        assetsToBackup.forEach { restoredAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(restoredAssetEntity.key).first()
+            assertTrue(asset != null)
+            assertEquals(asset, restoredAssetEntity)
+        }
+    }
+
+    @Test
+    fun givenUserHasAssetButBackupNot_whenRestoringBackup_thenUserAssetAreRestored() = runTest {
+        // given
+        val assetsToRestore = userDatabaseDataGenerator.generateAndInsertAssets(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        assetsToRestore.forEach { restoredAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(restoredAssetEntity.key).first()
+
+            assertTrue(asset != null)
+            assertEquals(asset, restoredAssetEntity)
+        }
+    }
+
+    @Test
+    fun givenUserHasUniqueAssetsAlongWithBackup_whenRestoringBackup_thenThoseAssetsAreRestored() = runTest {
+        // given
+        val assetsToRestore = userDatabaseDataGenerator.generateAndInsertAssets(10)
+        val backupAssets = backupDatabaseDataGenerator.generateAndInsertAssets(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        assetsToRestore.forEach { restoredAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(restoredAssetEntity.key).first()
+
+            assertTrue(asset != null)
+            assertEquals(asset, restoredAssetEntity)
+        }
+        backupAssets.forEach { backupAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(backupAssetEntity.key).first()
+
+            assertTrue(asset != null)
+            assertEquals(asset, backupAssetEntity)
+        }
+    }
+
+    private suspend fun insertOverlappingConversations(amount: Int): List<ConversationEntity> {
+        val conversationAdded = mutableListOf<ConversationEntity>()
+
+        for (index in 1..amount) {
+            val overLappingId = ConversationIDEntity("overlappingValue$index", "overlappingDomain$index")
+            val overLappingName = "overLappingName$index"
+
+            val conversationType = ConversationEntity.Type.values()[index % ConversationEntity.Type.values().size]
+
+            val invalidatedConversationType =
```

Comment:
I would either call this `sanitizedConversationType` or add the extra condition as a concatenated `let {}` on the line above to avoid confusion.